modules/highgui/include/opencv2/highgui.hpp

```diff
@@ -338,6 +338,21 @@ If there are several HighGUI windows, any of them can be active.
  */
 CV_EXPORTS_W int waitKey(int delay = 0);
 
+/** @brief Waits for a pressed key and returns the corresponding char value (ASCII) assigned.
+
+The function waitChar waits for a key event infinitely (when \f$\texttt{delay}\leq 0\f$ ) or for delay
+milliseconds, when it is positive. It returns the ASCII code of the pressed key as a char value or -1
```

Comment:
It's impossible to return -1 on systems with (char == unsigned char). I suppose return result became 255 for these systems.
Also there is a minor issue for (char == signed char) systems. Return value for timeout could be mixed up with value of valid `'\0xFF'` symbol(suppose it is an extremely rare usecase).